go.mod

```diff
@@ -1,6 +1,8 @@
 module github.com/parca-dev/parca-agent
 
-go 1.21.5
+go 1.21.1
+
+toolchain go1.21.4
```

Comment:
ah, any reason we went back to a previous version? 
maybe worth commenting why to avoid regressions if someone updates it

Reply:
The nix package doesn't have the latest Golang version. I can upgrade it to the latest version, and the go toolchain will download the missing tooling. However, we want reproducible builds, thus the downgrade. I believe these particular security fixes don't concern us.

Reply:
just to check I got it correct - nix doesn't have latest golang and we won't have reproducible builds without nix so we are using the previous version?

if so, and someone bumps the version just the same - do we have tests for reproducible builds (in this or another PR)?